src/libstd/thread/mod.rs

```diff
@@ -402,22 +410,19 @@ impl Builder {
             if let Some(name) = their_thread.cname() {
                 imp::Thread::set_name(name);
             }
-            unsafe {
-                thread_info::set(imp::guard::current(), their_thread);
-                #[cfg(feature = "backtrace")]
-                let try_result = panic::catch_unwind(panic::AssertUnwindSafe(|| {
-                    ::sys_common::backtrace::__rust_begin_short_backtrace(f)
-                }));
-                #[cfg(not(feature = "backtrace"))]
-                let try_result = panic::catch_unwind(panic::AssertUnwindSafe(f));
-                *their_packet.get() = Some(try_result);
-            }
+
+            thread_info::set(imp::guard::current(), their_thread);
```

Comment:
This is so sad, leaving the distinction between safe and unsafe code within this function just because it is unsafe. :(

Reply:
I am not sure what you mean. Are you concerned that there are more lines of safe code contained in a (function-scope) unsafe block?

Reply:
I am generally concerned about accidentally using unsafe operations inside an `unsafe fn`. Just because a function has an extra precondition doesn't mean it will recklessly want to use unsafe operations everywhere.

But that's just how the language is right now, and I'll need tow rite an RFC if I want it to change. Which I will do. Nothing you can do about it here.